src/main/java/org/openlmis/notification/web/usercontactdetails/UserContactDetailsController.java

```diff
@@ -169,6 +176,53 @@ public UserContactDetailsDto saveUpdateUserContactDetails(
     return toDto(contactDetails);
   }
 
+  /**
+   * Creates or updates contact details for multiple users.
+   *
+   * @param userContactDetailsDtoList  list of user contact details
+   * @return  the new or updated contact details for list of users
+   */
+  @PutMapping("/userContactDetails/batch")
+  @ResponseStatus(HttpStatus.OK)
+  @ResponseBody
+  public UserContactDetailsResponseDto saveUsersContactDetails(
+      @RequestBody List<UserContactDetailsDto> userContactDetailsDtoList) {
+    List<UserContactDetailsResponseDto.UserDetailsResponse> successfulResults = new ArrayList<>();
+    List<UserContactDetailsResponseDto.FailedUserDetailsResponse> failedResults = new ArrayList<>();
+
+    for (UserContactDetailsDto dto : userContactDetailsDtoList) {
+      try {
+        BindingResult bindingResult = new BeanPropertyBindingResult(dto, "userContactDetailsDto");
+        permissionService.canManageUserContactDetails(dto.getReferenceDataUserId());
+
+        validator.validate(dto, bindingResult);
+        List<String> errors;
+        if (bindingResult.hasErrors()) {
+          errors = bindingResult.getAllErrors().stream()
+              .map(DefaultMessageSourceResolvable::getDefaultMessage)
+              .collect(Collectors.toList());
+          failedResults.add(
+              new UserContactDetailsResponseDto.FailedUserDetailsResponse(
+                  dto.getReferenceDataUserId(), errors));
+        } else {
+          UserContactDetails contactDetails = UserContactDetails
+              .newUserContactDetails(dto);
+          userContactDetailsService.addOrUpdate(contactDetails);
+
+          successfulResults.add(
+              new UserContactDetailsResponseDto.UserDetailsResponse(dto.getReferenceDataUserId()));
+        }
+      } catch (Exception ex) {
+        failedResults.add(new UserContactDetailsResponseDto.FailedUserDetailsResponse(
+            dto.getReferenceDataUserId(),
+            Collections.singletonList(
+                MessageKeys.ERROR_SAVING_BATCH_CONTACT_DETAILS)));
```

Comment:
Could we include the exception's text in the error's text, maybe something like: `Something went wrong while saving contact details ({exception.toString()})`